numpy/ma/core.py

```diff
@@ -5412,35 +5427,22 @@ def sort(self, axis=-1, kind='quicksort', order=None,
         """
         if self._mask is nomask:
             ndarray.sort(self, axis=axis, kind=kind, order=order)
+            return
+
+        if self is masked:
+            return
+
+        sidx = self.argsort(axis=axis, kind=kind, order=order,
+                            fill_value=fill_value, endwith=endwith)
+
+        # save memory for 1d arrays
+        if self.ndim == 1:
+            idx = sidx
         else:
-            if self is masked:
-                return self
-            if fill_value is None:
-                if endwith:
-                    # nan > inf
-                    if np.issubdtype(self.dtype, np.floating):
-                        filler = np.nan
-                    else:
-                        filler = minimum_fill_value(self)
-                else:
-                    filler = maximum_fill_value(self)
-            else:
-                filler = fill_value
+            idx = list(np.ix_(*[np.arange(x) for x in self.shape]))
```

Comment:
Is this really a shortcut though? The two functions are independant, with neither calling the other